.github/workflows/pypi_release.yml

```diff
@@ -0,0 +1,63 @@
+name: Build and Release Python Bindings to PyPI
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+    inputs:
+      tag:
+        default: ''
+        required: false
+        description: Ref to build
+
+jobs:
+  build-sdist:
+    name: Build sdist
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repo 
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.tag }}
```

Comment:
does this just checkout master if no tag is provided?

Reply:
Actually, when checking out the repository that triggered a workflow, this defaults to the reference or SHA for that event.